desktop-widgets/diveplanner.cpp

```diff
@@ -232,10 +244,42 @@ void DivePlannerWidget::heightChanged(const int height)
 	plannerModel->setSurfacePressure(pressure);
 }
 
-void DivePlannerWidget::salinityChanged(const double salinity)
+void DivePlannerWidget::waterTypeUpdateTexts()
 {
-	/* Salinity is expressed in weight in grams per 10l */
-	plannerModel->setSalinity(lrint(10000 * salinity));
+	double density;
+	for (int i = 0; i < 3; i++) {
```

Comment:
For extensibility, please don't hardcode the number of entries. Use `ui.waterType->count()`.

Reply:
Ok.